controllers/nodehealthcheck_controller.go

```diff
@@ -412,6 +436,26 @@ func (r *NodeHealthCheckReconciler) remediate(ctx context.Context, node *v1.Node
 	return nextReconcile, nil
 }
 
+func (r *NodeHealthCheckReconciler) isControlPlaneRemediationAllowed(node *v1.Node, template *unstructured.Unstructured) (bool, error) {
+	if !utils.IsControlPlane(node) {
```

Comment:
IMO this is a bit confusing.
This is how I read it:

If node isn't  a control Plane node then Control Plane remediation is allowed.

I assume it's done so worker nodes  will go on with the normal remediation flow, but I suggest trying to rename it, so it would be clearer.

Reply:
I could remove the if statement here, because it's already at the calling place as well, but decided to leave it here just in case someone calls it for workers. I could also return an error... 🤷🏼‍♂️

Reply:
returning an error now